### PR TITLE
removed redundant messages and fix bugs

### PR DESCRIPTION
**Pull request by mayaraz**
- removed redundant messages every 10 minutes
- added log to `run_job` function
- fix two weeks bug (?)

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -47,28 +47,36 @@ def start(update: Update, context: CallbackContext) -> None:
...
+        job = context.job
+        # If it's less than 3 hours from now, kill it
+        if job.context.time - timedelta(hours=3) < (datetime.now()):
+            return
 
-    context.bot.send_message(job.context.chat_id, text=f'Job: {job.name}\n Could not find a court, '
-                                                       f'will try again in 10 minutes.')
-    context.job_queue.run_once(run_job, 60 * 10, context=job.context, name=job.name)
+        # If it's more than 2 weeks from now, we can't do anything till 2 weeks beforehand
+        if (job.context.time - datetime.now()).days > timedelta(weeks=2).days:
```
Comment (mayaraz) [inline bot.py:57]: Is this fixes the bug?
@moshekan

Comment (moshekan) [reply bot.py:57]: Seems like it

File: bot.py [hunk 2]
```diff
@@ -47,28 +47,36 @@ def start(update: Update, context: CallbackContext) -> None:
...
-        return
     try:
-        court = try_to_order(job.context.time, job.context.duration)
-        context.bot.send_message(job.context.chat_id, text=f'Successfully ordered court {court} for job {job.name}!')
-        return
-    except FailedToOrderCourtException:
-        pass
+        job = context.job
+        # If it's less than 3 hours from now, kill it
+        if job.context.time - timedelta(hours=3) < (datetime.now()):
```
Comment (moshekan) [inline bot.py:53]: I know that you haven't touched it, but worth reducing to 1 hour

File: bot.py [hunk 3]
```diff
@@ -47,28 +47,36 @@ def start(update: Update, context: CallbackContext) -> None:
...
+            return
+        try:
+            court = try_to_order(job.context.time, job.context.duration)
+            context.bot.send_message(job.context.chat_id,
+                                     text=f'Successfully ordered court {court} for job {job.name}!')
+            return
+        except FailedToOrderCourtException:
+            pass
+
+        logger.info(f"Job: {job.name}\n Could not find a court will try again in 10 minutes.")
```
Comment (moshekan) [inline bot.py:73]: Didn't you remove it?

**Review by moshekan — CHANGES_REQUESTED** [review]

File: bot.py (hunk 3 above)
Comment (mayaraz) [reply bot.py:73]: I log it now, instead of sending it

**Pull request merged**